Library/Homebrew/test/os/linux/software_spec_spec.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require "software_spec"
+
+describe SoftwareSpec do
+  subject(:spec) { described_class.new }
+
+  describe "#uses_from_macos" do
+    before do
+      allow(OS).to receive(:linux?).and_return(true)
+    end
+
+    it "allows specifying dependencies" do
+      spec.uses_from_macos("foo")
+
+      expect(spec.deps.first.name).to eq("foo")
+    end
+
+    it "works with tags" do
+      spec.uses_from_macos("foo" => :head, :after => :mojave)
+
+      expect(spec.deps.first.name).to eq("foo")
+      expect(spec.deps.first.tags).to include(:head)
+    end
+
+    it "ignores OS version specifications" do
+      spec.uses_from_macos("foo", after: :mojave)
+      spec.uses_from_macos("bar" => :head, :after => :mojave)
+
+      expect(spec.deps.first.name).to eq("foo")
+      expect(spec.deps.last.name).to eq("bar")
+      expect(spec.deps.last.tags).to include(:head)
+    end
+  end
+end
```

Comment:
Maybe it shouldn't be on `os/linux/` anyway, since it's the default behaviour

Reply:
If these tests fail on macOS, then they do need to be in `os/linux/`. If they pass on macOS, then yeah, they could be moved.

Reply:
You can use the `:needs_linux` tag instead for (the only) Linux specific tests. The test file folder locations should match the source file definitions.

Reply:
Mocking `OS.linux` shouldn't be enough for this case? It was running the extended `uses_from_macos` version even returning true on `OS.linux?`

Reply:
It would be sufficient but it's not how we've been doing things with other tests. Would appreciate consistency for now and could change elsewhere in a follow-up PR?